### ISSUE TITLE
Terraform wants to apply security list on every run, even with no changes

### ISSUE THREAD

**Issue by darkfibre** [oracle/terraform-provider-oci#565]
### Terraform Version

<details>

Terraform v0.11.7
+ provider.oci v2.1.16

</details>

### OCI Provider Version

<details>

terraform-provider-oci 2.1.16

</details>

### Description:
A security list that contains two ingress (or egress) rules that match the same address and port, but one matches the source port while the other matches the destination port, will result in a change being applied on every terraform run.

These rules predate the recent security list changes in the provider, and previously did not have any issues.

### Terraform Plan

<details>

These two rules are different, but terraform doesn't seem to recognize the existence of both of them when refreshing state, so it wants to apply a change on every run.

```
resource "oci_core_security_list" "test-seclist" {
    compartment_id = "${var.compartment_ocid}"
    display_name = "test-seclist"
    vcn_id = "${var.existing_vcn}"
    ingress_security_rules = [
      { source = "1.2.3.4/32" protocol = "6" tcp_options { source_port_range { "min" = 8443 "max" = 8443 } } stateless = true },
      { source = "1.2.3.4/32" protocol = "6" tcp_options { "min" = 8443 "max" = 8443 } stateless = true },
    ]
    egress_security_rules = [
    ]
}
```

No matter how many times you apply, the output of every run will be-

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ oci_core_security_list.test-seclist
      ingress_security_rules.#:                                                "1" => "2"
      ingress_security_rules.1780062820.icmp_options.#:                        "0" => "0"
      ingress_security_rules.1780062820.protocol:                              "6" => "6"
      ingress_security_rules.1780062820.source:                                "1.2.3.4/32" => "1.2.3.4/32"
      ingress_security_rules.1780062820.stateless:                             "true" => "true"
      ingress_security_rules.1780062820.tcp_options.#:                         "1" => "1"
      ingress_security_rules.1780062820.tcp_options.0.max:                     "0" => "0"
      ingress_security_rules.1780062820.tcp_options.0.min:                     "0" => "0"
      ingress_security_rules.1780062820.tcp_options.0.source_port_range.#:     "1" => "1"
      ingress_security_rules.1780062820.tcp_options.0.source_port_range.0.max: "8443" => "8443"
      ingress_security_rules.1780062820.tcp_options.0.source_port_range.0.min: "8443" => "8443"
      ingress_security_rules.1780062820.udp_options.#:                         "0" => "0"
      ingress_security_rules.2211539995.icmp_options.#:                        "0" => "0"
      ingress_security_rules.2211539995.protocol:                              "" => "6"
      ingress_security_rules.2211539995.source:                                "" => "1.2.3.4/32"
      ingress_security_rules.2211539995.source_type:                           "" => <computed>
      ingress_security_rules.2211539995.stateless:                             "" => "true"
      ingress_security_rules.2211539995.tcp_options.#:                         "0" => "1"
      ingress_security_rules.2211539995.tcp_options.0.max:                     "" => "8443"
      ingress_security_rules.2211539995.tcp_options.0.min:                     "" => "8443"
      ingress_security_rules.2211539995.tcp_options.0.source_port_range.#:     "0" => "0"
      ingress_security_rules.2211539995.udp_options.#:                         "0" => "0"


Plan: 0 to add, 1 to change, 0 to destroy.
```
The same issue exists with egress rules, if you have matching destinations/ports but you are matching on source vs destination port.

</details>

**Comment by alexng-canuck** [comment]
@darkfibre 
We are taking a look. 

The change plan seems to imply that the second security rule is being omitted from the state file after apply and so every subsequent apply is trying to add it back.

Does the second ingress rule actually get created when you apply (i.e. does it show up in console?)?

**Comment by darkfibre** [comment]
Yes, both rules exist before and after each terraform run.

**Comment by rcohenma** [comment]
This was fixed in release v2.2.0

https://github.com/oracle/terraform-provider-oci/releases/tag/v2.2.0
